File: 백준/Silver/17287. The Deeper， The Better/The Deeper， The Better.java
import java.io.*;
import java.util.*;

public class Main{
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        
        String line = br.readLine().trim();
            
        ArrayList<Integer> scoreBoard = new ArrayList<>();
        int score = 0;
        int result = 0;

        for (char ch : line.toCharArray()) {
            switch (ch) {
                case '(':
                    score += 1;
                    break;
                case '{':
                    score += 2;
                    break;
                case '[':
                    score += 3;
                    break;
                case ')':
                    score -= 1;
                    break;
                case '}':
                    score -= 2;
                    break;
                case ']':
                    score -= 3;
                    break;
                default:
                    if (score > result) result = score;
                    break;
            }
        }

        bw.write(String.valueOf(result));

		bw.flush();
		bw.close();
		br.close();
	}
}

File: 백준/Silver/17287. The Deeper， The Better/The Deeper， The Better.py
import sys

input = sys.stdin.readline

line = input().strip()
score = 0
score_board = []
result = 0

for li in line:
    if li == '(':
        score += 1
    elif li == '{':
        score += 2
    elif li == '[':
        score += 3
    elif li == ')':
        score -= 1
    elif li == '}':
        score -= 2
    elif li == ']':
        score -= 3
    else:
        score_board.append((li, score))

for _, score in score_board:
    if result < score:
        result = score

print(result)